library/src/com/google/maps/android/data/MultiGeometry.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.maps.android.data;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * An abstraction that shares the common properties of
+ * {@link com.google.maps.android.data.kml.KmlMultiGeometry KmlMultiGeometry}
+ * and {@link com.google.maps.android.data.geojson.GeoJsonMultiLineString GeoJsonMultiLineString},
+ * {@link com.google.maps.android.data.geojson.GeoJsonMultiPoint GeoJsonMultiPoint} and
+ * {@link com.google.maps.android.data.geojson.GeoJsonMultiPolygon GeoJsonMultiPolygon}
+ */
+public class MultiGeometry implements Geometry {
+
+    private String geometryType = "MultiGeometry";
+
+    private List<Geometry> mGeometries;
+
+    /**
+     * Creates a new MultiGeometry object
+     * @param geometries contains list of Polygons, Linestrings or Points
+     */
+    public MultiGeometry(List<?> geometries) {
```

Comment:
Unless I'm missing something, from a quick look through the uses of this class it seems like objects passed in must extend `Geometry`.  IMHO a better approach for the implementation here, assuming we want to copy the provided list and not just keep a reference to it, would be:

~~~
    public MultiGeometry(List<? extends Geometry> geometries) {
        if (geometries == null) {
            throw new IllegalArgumentException("Geometries cannot be null");
        }
        mGeometries = new ArrayList<>(geometries);
    }
~~~

Simpler, doesn't assume objects extend `Geometry`, and allows platform to optimize the copying of the `ArrayList`.